core/src/area_data.cpp

```diff
@@ -504,3 +523,49 @@ void AreaData::toggleIgnoreBgList()
 {
     m_ignoreBgList = !m_ignoreBgList;
 }
+
+void AreaData::toggleJukebox()
+{
+    m_jukebox = !m_jukebox;
+    if (!m_jukebox) {
+        m_jukebox_queue.clear();
+        m_jukebox_timer->stop();
+    }
+}
+
+bool AreaData::addJukeboxSong(QString f_song)
+{
+    if(!m_jukebox_queue.contains(f_song)) {
+        if (m_jukebox_queue.size() == 0) {
+            emit playJukeboxSong(AOPacket("MC",{f_song,QString::number(-1)}), index());
+            m_jukebox_timer->start(ConfigManager::songInformation(f_song) * 1000);
+            setCurrentMusic(f_song);
+            setMusicPlayedBy("Jukebox");
+        }
+        m_jukebox_queue.append(f_song);
+        return true;
+    }
+    else {
+        return false;
+    }
+}
+
+void AreaData::switchJukeboxSong()
+{
+    QString l_song_name;
+    if(m_jukebox_queue.size() == 1) {
+        l_song_name = m_jukebox_queue[0];
+        emit playJukeboxSong(AOPacket("MC",{l_song_name,QString::number(-1)}), m_index);
```

Comment:
```suggestion
        emit playJukeboxSong(AOPacket("MC",{l_song_name,"-1"}), m_index);
```
just use a string literal pls